workflows/readmapping.nf

```diff
@@ -63,86 +58,101 @@ workflow READMAPPING {
 
     ch_versions = Channel.empty()
 
+
     //
     // SUBWORKFLOW: Read in samplesheet, validate and stage input files
     //
-    INPUT_CHECK ( ch_input )
-        .reads
-        .branch {
-            meta, reads ->
-                hic : meta.datatype == "hic"
-                    return [ meta, reads ]
-                illumina : meta.datatype == "illumina"
-                    return [ meta, reads ]
-                pacbio : meta.datatype == "pacbio"
-                    return [ meta, reads ]
-                clr : meta.datatype == "pacbio_clr"
-                    return [ meta, reads ]
-                ont : meta.datatype == "ont"
-                    return [ meta, reads ]
-        }
-        .set { ch_reads }
-    ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)
+    INPUT_CHECK ( ch_input ).reads
+    | branch {
+        meta, reads ->
+            hic : meta.datatype == "hic"
+            illumina : meta.datatype == "illumina"
+            pacbio : meta.datatype == "pacbio"
+            clr : meta.datatype == "pacbio_clr"
+            ont : meta.datatype == "ont"
+    }
+    | set { ch_reads }
+
+    ch_versions = ch_versions.mix ( INPUT_CHECK.out.versions )
+
 
     //
     // SUBWORKFLOW: Uncompress and prepare reference genome files
     //
-    PREPARE_GENOME ( ch_fasta )
-    ch_versions = ch_versions.mix(PREPARE_GENOME.out.versions)
+    ch_fasta
+    | map { [ [ id: it.baseName.tokenize(".")[0..1].join(".") ], it ] }
+    | set { ch_genome }
+
+    PREPARE_GENOME ( ch_genome )
+    ch_versions = ch_versions.mix ( PREPARE_GENOME.out.versions )
+
 
     //
     // Create channel for vector DB
     //
-    if (params.vector_db.endsWith('.tar.gz')) {
-        ch_db   = UNTAR ([ [:], params.vector_db ]).untar.map { meta, file -> file }
-        ch_versions = ch_versions.mix(UNTAR.out.versions)
-    } else {
-        ch_db   = file(params.vector_db)
+    // ***PacBio condition does not work - needs fixing***
```

Comment:
What is the problem ?

Reply:
🤦‍♀️ I forgot to delete that comment, it works now.